File: backend/app/schemas/notification.py
from pydantic import BaseModel
from typing import Optional
from datetime import datetime


class NotificationBase(BaseModel):
    type: str
    level: str = "info"
    title: str
    content: Optional[str] = None
    recipient: Optional[str] = None
    channel: str = "in_app"


class NotificationCreate(NotificationBase):
    pass


class NotificationUpdate(BaseModel):
    status: Optional[str] = None


class NotificationResponse(NotificationBase):
    id: int
    status: str = "pending"
    sent_at: Optional[datetime] = None
    read_at: Optional[datetime] = None
    created_at: Optional[datetime] = None

    class Config:
        from_attributes = True

File: frontend/src/types/notification.ts
// 字段与后端 schemas/notification.py::NotificationResponse 对齐
export interface NotificationItem {
  id: number;
  type: string;
  level: string; // info / warning / error
  title: string;
  content?: string;
  recipient?: string;
  channel: string;
  status: string; // pending / sent / read
  sent_at?: string;
  read_at?: string;
  created_at?: string;
}
